sphire/convert.py

```diff
@@ -131,30 +131,44 @@ def __init__(self, boxSize, yFlipHeight=None, boxSizeEstimated=False):
         self._boxSizeEstimated = boxSizeEstimated
 
     def iterCoords(self, filename):
-        for row in Table.iterRows(filename, tableName='cryolo'):
-            x = row.CoordinateX
-            y = row.CoordinateY
-            z = row.get("CoordinateZ", 0.0)
-            score = row.get("Confidence", 0.0)
-            groupId = int(row.get('filamentid', 0))
-            width = int(row.get('Width', 0))
-
-            if self._boxSize is None:
-                self._boxSize = width
-                self._halfBox = self._boxSize / 2.0
-
-            if not self._boxSizeEstimated:
-                x += self._halfBox
-                y += self._halfBox
-
-            sciX = round(x)
-            sciY = round(y)
-            sciZ = round(z) if not isinstance(z, str) else 0  # avoid <NA> values
-
-            if self._yFlipHeight is not None:
-                sciY = self._yFlipHeight - sciY
-
-            yield sciX, sciY, sciZ, score, groupId, width
+        ext = pwutils.getExt(filename)
+        if ext == '.cbox':
+            for row in Table.iterRows(filename, tableName='cryolo'):
+                x = row.CoordinateX
+                y = row.CoordinateY
+                z = row.get("CoordinateZ", 0.0)
+                score = row.get("Confidence", 0.0)
+                groupId = int(row.get('filamentid', 0))
+                width = int(row.get('Width', 0))
+
+                if self._boxSize is None:
+                    self._boxSize = width
+                    self._halfBox = self._boxSize / 2.0
+
+                if not self._boxSizeEstimated:
+                    x += self._halfBox
+                    y += self._halfBox
+
+                sciX = round(x)
+                sciY = round(y)
+                sciZ = round(z) if not isinstance(z, str) else 0  # avoid <NA> values
+
+                if self._yFlipHeight is not None:
+                    sciY = self._yFlipHeight - sciY
+
+                yield sciX, sciY, sciZ, score, groupId, width
+
+        elif ext == '.coords':
+            with open(filename, 'r') as file:
+                lines = file.readlines()
+                for line in lines:
+                    values = line.strip().split(' ')
```

Comment:
Is it a type or x, y, z are all taken from values[0]?

Reply:
.coords files are flat text files where they only have rows with the x, y and z coordinates separated by ' '. It is a file format that napari-boxmanager generates as well. e.g.

208.75713019549423 72.95971762870192 25.0
198.63182438188855 49.13546865551214 25.0
174.80757540869877 35.43652549592801 25.0

Reply:
yes @delarosatrevin it looks like a typo, otherwise all coords are the same value